File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectModel/src/main/java/ro/mpp2025/javaprojectui/Inscriere.java
package ro.mpp2025.javaprojectui;
import java.util.Objects;

public class Inscriere extends Entity<Tuplu<Participant, Proba>> {
    private Participant IDparticipant;
    private Proba IDproba;

    public Inscriere(Participant IDparticipant, Proba IDproba) {
        this.IDparticipant = IDparticipant;
        this.IDproba = IDproba;
        this.setId(new Tuplu<>(IDparticipant, IDproba));
    }

    public Participant getIDparticipant() {
        return IDparticipant;
    }

    public Proba getIDproba() {
        return IDproba;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Inscriere inscriere = (Inscriere) o;
        return Objects.equals(IDparticipant, inscriere.IDparticipant) && Objects.equals(IDproba, inscriere.IDproba);
    }

    @Override
    public int hashCode() {
        return Objects.hash(IDparticipant, IDproba);
    }
}

File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Model/Inscriere.cs
using System.Text.Json.Serialization;

namespace Model;
[Serializable]
public class Inscriere : Entity<System.Tuple<Participant, Proba>>
{
    private Participant _idParticipant;
    private Proba _idProba;
        
    public Inscriere(){}
    [JsonConstructor]
    public Inscriere(Participant idParticipant, Proba idProba)
    {
        this._idParticipant = idParticipant;
        this._idProba = idProba;
        this.Id = new Tuple<Participant, Proba>(idParticipant, idProba);
    }

    public override string ToString()
    {
        return
            $"{nameof(_idParticipant)}: {_idParticipant}, {nameof(_idProba)}: {_idProba}, {nameof(IDParticipant)}: {IDParticipant}, {nameof(IDProba)}: {IDProba}";
    }

    public Participant IDParticipant
    {
        get { return _idParticipant; }
        set { _idParticipant = value; }
    }
        
    public Proba IDProba
    {
        get { return _idProba; }
        set { _idProba = value; }
    }
        
    public override bool Equals(object obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;
            
        Inscriere inscriere = (Inscriere)obj;
        return _idParticipant == inscriere._idParticipant && _idProba == inscriere._idProba;
    }
        
    public override int GetHashCode()
    {
        return HashCode.Combine(_idParticipant, _idProba);
    }
}